### PR TITLE
feat(ci): implement release task collection from GitHub API

### LINKED ISSUE #101: [release-management] Collect release tasks from GitHub using labels and milestone
## Context

Once the current release has been identified from the `README.md`, the automation needs to determine which tasks belong to that release.

The release changelog should not be generated from commits. Instead, it should be based on GitHub Issues/Tasks, using repository planning as the source of truth.

Only tasks that:
- have the `release` label
- belong to the milestone matching the current release name

should be considered for changelog generation.

Without this filtering step, the changelog may include unrelated work or miss work that belongs to the current release.

## Objective

Implement logic to collect GitHub tasks associated with the current release by filtering open issues using:
- label: `release`
- milestone: current release name

### Included
- Receive current release metadata from the release detection step
- Query GitHub Issues/Tasks
- Filter only issues with label `release`
- Filter only issues whose milestone matches the current release name
- Return a structured list of release tasks for downstream processing

### Excluded
- Parsing the README release table
- Classifying changelog entries into Added/Changed/Fixed/Removed
- Writing `CHANGELOG.md`
- CI trigger conditions

### Scenario 1 — Happy Path
**Given**  
the current release name has been identified  
and GitHub contains open tasks labeled `release` with a matching milestone  

**When**  
the task collection logic runs  

**Then**  
only tasks matching both the label and milestone criteria are returned  

### Scenario 2 — Edge Case
**Given**  
some open tasks have the `release` label but belong to a different milestone  
and some tasks belong to the milestone but do not have the `release` label  

**When**  
the task collection logic runs  

**Then**  
those tasks are excluded  
and only tasks satisfying both filters are collected  

### Scenario 3 — Failure Case
**Given**  
no open tasks match the current release name and `release` label  

**When**  
the task collection logic runs  

**Then**  
the workflow returns an explicit empty result or raises a controlled error  
according to the defined changelog generation strategy  

## Acceptance Criteria

- [x] Only issues labeled `release` are considered
- [x] Only issues with milestone equal to the current release name are considered
- [x] The result is structured and reusable by downstream changelog generation steps
- [x] Non-matching issues are excluded deterministically
- [x] Tests cover happy path, edge case, and failure case scenarios

## Notes (Optional)

The release name used for milestone matching comes from the `README.md` release roadmap.

The output of this task should include enough metadata for consolidation later, such as:
- issue number
- title
- labels
- milestone
- body
- state

## Related Documents

- docs/01-requirements/constraints.md
- docs/02-architecture/overview.md

## Related Issue

Closes #

### PR DESCRIPTION
**Pull request by brunolnetto**
## Changes

- Add `tools/changelog_generator/task_collector.py` — `TaskCollector` queries the GitHub REST API to find the milestone matching the current release name, then fetches all open issues in that milestone with the `release` label; returns a list of `ReleaseTask` frozen dataclasses
- Add `tools/changelog_generator/tests/test_task_collector.py` — unit tests using a fake `urlopen` to avoid real HTTP calls; covers happy path, empty results, missing milestone, HTTP 401, and URLError

## Motivation

The changelog generation pipeline (#103) needs to collect the set of release tasks that carry changelog entries. This PR implements the second step: querying GitHub's API for issues that belong to the current release milestone and are tagged with the `release` label.

## Impact

- Builds on `ReleaseMetadata` from PR #100 (feature/task-100-release-detection)
- No new runtime dependencies (uses `urllib.request` from stdlib)
- Requires `GITHUB_TOKEN` at runtime; never logged or exposed

## Checklist

- [x] Tests added for new code
- [x] Uses stdlib only (`urllib.request`)
- [x] Token passed via constructor argument — not hardcoded
- [x] Commit message follows the convention

## Related Issue

Closes #101

### CODE REVIEW

File: tools/changelog_generator/tests/test_release_detector.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+## Releases & Roadmap
+
+Release Roadmap:
+
+| Version | Name | Status | Planned Date | Details |
+|--------|------|--------|---------|---------|
+| v0.2.1 | Old Release | ✅ Released | 2026-03-01 | [Notes](https://example.com/notes) |
+| v0.2.2 | Curriculum Graph API Foundation | 🚧 In Progress | 2026-06-15 | [Plan](https://example.com/186) |
+| v0.3.0 | Orchestrator Core | 🔜 Planned | 2026-09-01 | [Plan](https://example.com/185) |
```
[the remark this answers is not in the document]
Comment (brunolnetto) [reply tools/changelog_generator/tests/test_release_detector.py:26]: @copilot apply changes based on [this feedback](https://github.com/AigoraCorporation/aigora/pull/224#discussion_r3267055975)

File: tools/changelog_generator/task_collector.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                f"No GitHub milestone found matching release name: {release.name!r}"
+            )
+        return self._fetch_tasks(milestone_number)
+
+    def _find_milestone_number(self, name: str) -> int | None:
+        milestones = self._get(
+            f"/repos/{self._owner}/{self._repo}/milestones?per_page=100&state=all"
+        )
+        return next(
+            (ms["number"] for ms in milestones if ms["title"] == name),
```
[the remark this answers is not in the document]
Comment (brunolnetto) [reply tools/changelog_generator/task_collector.py]: @copilot apply changes based on [this feedback](https://github.com/AigoraCorporation/aigora/pull/224#discussion_r3267056041)

File: tools/changelog_generator/task_collector.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            None,
+        )
+
+    def _fetch_tasks(self, milestone_number: int) -> list[ReleaseTask]:
+        path = (
+            f"/repos/{self._owner}/{self._repo}/issues"
+            f"?milestone={milestone_number}&labels=release&state=open&per_page=100"
+        )
+        return [self._map_task(issue) for issue in self._get(path)]
```
[the remark this answers is not in the document]
Comment (brunolnetto) [reply tools/changelog_generator/task_collector.py]: @copilot apply changes based on [this feedback](https://github.com/AigoraCorporation/aigora/pull/224#discussion_r3267056070)

File: tools/changelog_generator/release_detector.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass
+from pathlib import Path
```
[the remark this answers is not in the document]
Comment (brunolnetto) [reply tools/changelog_generator/release_detector.py:5]: @copilot apply changes based on [this feedback](https://github.com/AigoraCorporation/aigora/pull/224#discussion_r3267056253)

File: tools/changelog_generator/task_collector.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            url,
+            headers={
+                "Authorization": f"Bearer {self._token}",
+                "Accept": "application/vnd.github+json",
+                "X-GitHub-Api-Version": "2022-11-28",
+            },
+        )
+        try:
+            with urllib.request.urlopen(req) as response:
+                return json.loads(response.read().decode("utf-8"))
```
[the remark this answers is not in the document]
Comment (brunolnetto) [reply tools/changelog_generator/task_collector.py:108]: @copilot apply changes based on [this feedback](https://github.com/AigoraCorporation/aigora/pull/224#discussion_r3267056302)

**Comment by brunolnetto** [comment]
@copilot resolve the merge conflicts in this pull request
